spec/core_ext/class/hierarchy_spec.rb

```diff
@@ -1,6 +1,24 @@
 require 'socket' # for a more interesting hierarchy for specs
 
 describe Class do
+  context "#descendant_get" do
+    it "returns self if specified" do
+      expect(IO.descendant_get("IO")).to eq(IO)
+    end
+
+    it "returns a direct descendant" do
+      expect(IO.descendant_get("BasicSocket")).to eq(BasicSocket)
+    end
+
+    it "returns a transitive descendant" do
+      expect(IO.descendant_get("IPSocket")).to eq(IPSocket)
+    end
+
+    it "raises an error if no descendant is found" do
+      expect { IO.descendant_get("Foo") }.to raise_error(ArgumentError, "Foo is not a descendant of IO")
```

Comment:
Should this be testing a constant that exists, but is not a descendant of IO?  Like `"MoreCoreExtensions"`?

Reply:
Good one...I'll add a test for that.